### PR TITLE
feat(toolkit-cleaner): clean unused S3 and ECR assets from CDK Toolkit

### PR DESCRIPTION
**Pull request by jogold**
The `ToolkitCleaner` construct creates a state machine that runs every day
and removes unused S3 and ECR assets from your CDK Toolkit.

It starts by getting all stack names. For each stack, it gets its template
and extracts unique assets hashes. The content of the toolkit bucket and
repository is then compared to those hashes and unused assets are removed. 

State machine:

### CODE REVIEW

**Comment by jogold** [comment]
@eladb would love to have your feedback here!

**Comment by eladb** [comment]
Super cool! @kaizen3031593 would be interested in this as he has been working on a related RFC: https://github.com/aws/aws-cdk-rfcs/pull/379

File: src/toolkit-cleaner/index.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    // Dummy assets to reference S3 bucket and ECR repository
+    const fileAsset = new Asset(this, 'FileAsset', {
+      path: path.join(__dirname, '..', '..', 'assets', 'toolkit-cleaner', 'docker', 'dummy.txt'),
+    });
+    const dockerImageAsset = new DockerImageAsset(this, 'DockerImageAsset', {
+      directory: path.join(__dirname, '..', '..', 'assets', 'toolkit-cleaner', 'docker'),
+    });
+
+    const getStackNamesHandler = new NodejsFunction(this, 'get-stack-names');
```
Comment (eladb) [inline src/toolkit-cleaner/index.ts:40]: Curious - why are you not using projen's lambda support to handle pre-bundling...

Comment (jogold) [reply src/toolkit-cleaner/index.ts:46]: Because I forgot about it :)

**Review by eladb — COMMENTED** [review]
Looks awesome

**Pull request merged**